x/interchainstaking/keeper/callbacks.go

```diff
@@ -454,6 +454,12 @@ func DepositTxCallback(k *Keeper, ctx sdk.Context, args []byte, query icqtypes.Q
 		return err
 	}
 
+	// check client state validity
+	err = k.CheckTMHeaderForZone(ctx, &zone, res)
+	if err != nil {
+		return err
+	}
```

Comment:
I think we should move these lines to start from the line `451` after we call `keeper.GetReceipt`